IOPool/Output/src/RootOutputFile.cc

```diff
@@ -696,6 +694,27 @@ namespace edm {
 
     std::set<StoredProductProvenance> provenanceToKeep;
 
+    ProductProvenanceRetriever const* provRetriever = nullptr;
+    {
+      EventPrincipal const* eventPrincipal = dynamic_cast<EventPrincipal const*>(&principal);
+      if (eventPrincipal) {
+        provRetriever =eventPrincipal->productProvenanceRetrieverPtr();
+        assert(provRetriever);
+      }
+    }
+    
+    //If we are dropping some of the meta data we need to know
```

Comment:
@Dr15Jones I don't understand the purpose of lines 707-717, and the use of "producedBranches" after that.